### ISSUE TITLE
Bootstrap 5

### ISSUE THREAD

**Issue by Clashsoft** [Clashsoft/bootstrap-darkmode#23]
While the CSS mostly still works with Bootstrap 5, the following things may need to be changed (taken from the [Migration Guide](https://getbootstrap.com/docs/5.0/migration)):

## Content, Reboot, etc.

- [x] Redesigned tables to refresh their styles and rebuild them with CSS variables for more control over styling.
- [x] .thead-light and .thead-dark are dropped in favor of the .table-* variant classes which can be used for all table elements (thead, tbody, tfoot, tr, th and td).
- [x] .text-* utilities do not add hover and focus states to links anymore. .link-* helper classes can be used instead.

## Forms

- [x] Consolidated native and custom form elements. Checkboxes, radios, selects, and other inputs that had native and custom classes in v4 have been consolidated. Now nearly all our form elements are entirely custom, most without the need for custom HTML.
  - [x] .custom-check is now .form-check.
  - [x] .custom-check.custom-switch is now .form-check.form-switch.
  - [x] .custom-select is now .form-select.
  - [x] .custom-file and .form-file have been replaced by custom styles on top of .form-control.
  - [x] .custom-range is now .form-range.
  - [x] Dropped native .form-control-file and .form-control-range.
- [x] Dropped .input-group-append and .input-group-prepend. You can now just add buttons and .input-group-text as direct children of the input groups.

## Accordion
- [x] Added new accordion component.

## Alerts
- [x] Removed custom styles for `<hr>`s in each alert since they already use currentColor.

## Badges
- [x] Dropped all .badge-* color classes for background utilities (e.g., use .bg-primary instead of .badge-primary).
- [x] Removed hover and focus styles for `<a>` and `<button>` elements.

## Breadcrumbs
- [x] Simplified the default appearance of breadcrumbs by removing padding, background-color, and border-radius.

## Carousel
- [x] Added new .carousel-dark variant for dark text, controls, and indicators (great for lighter backgrounds).

## Close button
- [x] Renamed .close to .btn-close for a less generic name.
- [x] Close buttons now use a background-image (embedded SVG) instead of a `&times;` in the HTML, allowing for easier customization without the need to touch your markup.

## Dropdowns
- [x] Added new .dropdown-menu-dark variant and associated variables for on-demand dark dropdowns.
- [x] Darkened the dropdown divider for improved contrast.

## Jumbotron
- [x] Dropped the jumbotron component as it can be replicated with utilities. See our new Jumbotron example for a demo.

## Popovers
- [x] Renamed .arrow to .popover-arrow in our default popover template.

## Utilities
- [x] Added new .bg-body class for quickly setting the `<body>`’s background to additional elements.

**Comment by slaweet** [comment]
There's also some `sass` stuff that need updating. When I tried to use `@import "~bootstrap-darkmode/scss/darktheme";` with boostrap v5, it fails on:
```
Failed to compile.

./src/components/Layout/AfterLoginStyles.vue?vue&type=style&index=2&lang=scss& (./node_modules/css-loader/dist/cjs.js??ref--8-oneOf-1-1!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/postcss-loader/src??ref--8-oneOf-1-2!./node_modules/sass-loader/dist/cjs.js??ref--8-oneOf-1-3!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/Layout/AfterLoginStyles.vue?vue&type=style&index=2&lang=scss&)
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: $color2: theme-color-level("primary", -6) is not a color.
    ╷
206 │   @return mix(black, $color, $weight);
    │           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
  node_modules/bootstrap/scss/_functions.scss 206:11                 shade-color()
  node_modules/bootstrap/scss/mixins/_alert.scss 8:12                alert-variant()
  node_modules/bootstrap-darkmode/scss/mixins/_alert.scss 4:4        dark-alert()
  node_modules/bootstrap-darkmode/scss/mixins/_dark-theme.scss 57:2  dark-theme()
  node_modules/bootstrap-darkmode/scss/darktheme.scss 9:3            @import
  src/components/Layout/AfterLoginStyles.vue 62:9                    root stylesheet
```

According to https://getbootstrap.com/docs/5.0/migration/#sass:

> Renamed theme-color-level() function to color-level() and now accepts any color you want instead of only $theme-color colors. See #29083 Watch out: color-level() was later on dropped in v5.0.0-alpha3.

**Comment by Clashsoft** [comment]
@slaweet Thanks for documenting. Due to the number of breaking changes, I will probably make a separate branch / version. E.g. `bootstrap-darkmode@^4` for `bootstrap@^4` and `bootstrap-darkmode@^5` for `bootstrap@^5`.

**Comment by jadeops** [comment]
Hi, any ETA for Bootstrap v5 dark mode release?

**Comment by Clashsoft** [comment]
Unfortunately no ETA but I will start working on it. The issue description now has a list of things that may need to be done, filtered from the Migration Guide to only the relevant items.

I encourage everyone to try the ~~v4~~ latest version of the library anyway and see what most obviously breaks so I can prioritise.

**Comment by Clashsoft** [comment]
I published ~~`0.5.0-alpha.1`~~ ~~`0.5.0-alpha.2`~~  ~~`0.5.0-alpha.3`~~  [`0.5.0-alpha.4`](https://www.npmjs.com/package/bootstrap-darkmode/v/5.0.0-alpha.4) with all the SCSS errors fixed. That is, with all the erroneous SCSS simply removed, for now. Many things will likely break or look off. I will start with the most important things -

- [x] links
- [x] buttons
- [x] colored links
- [x] forms
- [x] dropdowns
- [x] alerts
- [x] list groups
- [x] navs
- [x] tables

**Comment by Clashsoft** [comment]
[`5.0.0-beta.1`](https://www.npmjs.com/package/bootstrap-darkmode/v/5.0.0-beta.1) is now out with most things taken care of. Let me know if you encounter any oddities.
